akka-contrib/src/main/scala/akka/contrib/throttle/TimerBasedThrottler.scala

```diff
@@ -295,10 +300,30 @@ class TimerBasedThrottler(var rate: Rate) extends Actor with FSM[State, Data] {
    */
   private def deliverMessages(data: Data): Data = {
     val queue = data.queue
-    val nrOfMsgToSend = scala.math.min(queue.length, data.callsLeftInThisPeriod)
 
-    queue.take(nrOfMsgToSend).foreach(x ⇒ data.target.get.tell(x.message, x.sender))
+    import akka.pattern.ask
+    implicit val timeout = Timeout(61.seconds)
+    val callsInThisPeriod =
+      Try(Await.result((stopwatch ? LastPeriodCount(rate.duration)).mapTo[LastPeriodCountResponse], 60.seconds)) match {
```

Comment:
Blocking Await is not nice here, I think we'll close this without merging. Thanks anyway. Shouldn't be a problem to maintain a copy of this in your project.